docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,698 @@
+
+> **All share commitments included in `MsgPayForData` must consist only of subtree roots of the data square.**
+
+The main issue with that requirement is that users must know the relevant subtree roots before they sign, which is problematic considering that if the block is not organized perfectly, the subtree roots will include data unknown to the user at the time of signing.
+
+To fix this, the spec outlines the “non-interactive default rules”. These involve a few additional **default but optional** message layout rules that enables the user to follow the above block validity rule, while also not interacting with a block producer. Commitments to messages can consist entirely of sub tree roots of the data hash, and for those sub tree roots to be generated only from the message itself (so that the user can sign something “non-interactively”).
+
+NOTE: THE STATEMENT BELOW IS MODIFIED FROM THE SPEC
+
+> **Messages must begin at a location aligned with the largest power of 2 that is not larger than the message length or k.**
+
+Below illustrates how we can break a message up into two different subtree roots, following a [Merkle Mountain Range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure, i.e. root `A` for first four shares, and root `B` consisting of the last two shares. We can then create a commitment out of any number of subtree roots by creating a merkle root of those commitments.
+
+![Subtree root commitment](./assets/subtree-root.png "Subtree Root based commitments")
+
+> **If the messages are larger than k, then they must start on a new row.**
+
+![Multi row share commitment](./assets/multirow.png "Subtree Root based commitments")
+
+If we can follow these rules, we can always create a commitment to the data that is a subtree root of the data root while only knowing the data in that message.
+
+In practice this usually means that we end up adding padding between messages (zig-zag hatched share) to ensure that each message is starting at an "aligned power of two". Padding consists of shares with the namespace of the message before it, with all zeros for data.
+
+![Before using the non-interactive defaults](./assets/before.png "Before requiring that all commits consist of subtree roots")
+![after](./assets/after.png "After requiring that all commits consist of subtree roots")
+
+Below is an example block that has been filled using the non-interactive default rules to ensure that all commitments consist only of subtree roots.
+
+![example](./assets/example-full-block.png "example")
+
+- `ns = 1`: The first namespace is put in the beginning of the messages space.
+- `ns = 2`: Starting at the largest power of 2 that is not  (strictly) larger than the message length or `k`.
+- `ns = 3`: Message spanning multiple rows. So, it will be at the beginning of a new row.
+- `ns  = 4`: Starting at the largest power of 2 that is not  (strictly) larger than the message length or `k`.
```

Comment:
```suggestion
- `ns = 2`: Starting at the largest power of 2 that is not larger than the message length or `k`.
- `ns = 3`: Message spanning multiple rows. So, it will be at the beginning of a new row.
- `ns  = 4`: Starting at the largest power of 2 that is not larger than the message length or `k`.
```